google/cloud/forseti/scanner/scanners/location_scanner.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2018 The Forseti Security Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Scanner for the Lien rules engine."""
+
+from google.cloud.forseti.common.gcp_type import bucket
+from google.cloud.forseti.common.gcp_type import project
+from google.cloud.forseti.common.util import logger
+from google.cloud.forseti.scanner.audit import location_rules_engine
+from google.cloud.forseti.scanner.scanners import base_scanner
+
+
+LOGGER = logger.get_logger(__name__)
+
+
+class LocationScanner(base_scanner.BaseScanner):
+    """Scanner for Liens."""
+
+    def __init__(self, global_configs, scanner_configs, service_config,
+                 model_name, snapshot_timestamp,
+                 rules):
+        """Initialization.
+        Args:
+            global_configs (dict): Global configurations.
+            scanner_configs (dict): Scanner configurations.
+            service_config (ServiceConfig): Forseti 2.0 service configs
+            model_name (str): name of the data model
+            snapshot_timestamp (str): Timestamp, formatted as YYYYMMDDTHHMMSSZ.
+            rules (str): Fully-qualified path and filename of the rules file.
+        """
+        super(LocationScanner, self).__init__(
+            global_configs,
+            scanner_configs,
+            service_config,
+            model_name,
+            snapshot_timestamp,
+            rules)
+        self.rules_engine = location_rules_engine.LocationRulesEngine(
+            rules_file_path=self.rules,
+            snapshot_timestamp=self.snapshot_timestamp)
+        self.rules_engine.build_rule_book(self.global_configs)
+
+    def run(self):
+        """Runs the data collection."""
+        resources = self._retrieve()
+        all_violations = self._find_violations(resources)
+        self._output_results(all_violations)
+
+    def _retrieve(self):
+        """Retrieves the data for scanner.
+
+        Returns:
+            List[Resource]: resources to check for violations.
+        Raises:
+            ValueError: if resources have an unexpected type.
+        """
+        resources = []
+
+        resource_type_to_fn = {'bucket': bucket.Bucket.from_json}
```

Comment:
Document what kind of function is expected here. Perhaps we could move this this out to somewhere the rule engine and scanner can both see, so there is a single source of truth for supported resources

Reply:
Done.